src/lib/transaction_pool/transaction_pool.ml

```diff
@@ -20,10 +28,43 @@ module type Transition_frontier_intf = sig
     end
   end
 
+  val best_tip : t -> Breadcrumb.t
+
   val best_tip_diff_pipe :
     t -> Extensions.Best_tip_diff.view Broadcast_pipe.Reader.t
 end
 
+module type User_command_intf = sig
+  type t [@@deriving sexp, bin_io]
```

Comment:
See https://github.com/CodaProtocol/coda/blob/master/rfcs/0015-module-versioning.md , `bin_io` shouldn't be used directly on `type t` anymore.

Would be nice if we could just directly depend on `user_command` but I guess it's in coda_base?

Reply:
ping @psteckler if you have any questions on how to stablify it

Reply:
We can just depend on user command directly, but it's harder to test, hence the functoring. The `Make` functor doesn't take a user command parameter, but `Make0` does.

Reply:
I made it use `User_command.Stable.Latest.t`.